plugins/builds/index.js

```diff
@@ -812,115 +854,165 @@ exports.register = (server, options, next) => {
-                parentBuilds
-            };
-            const externalBuildConfig = {
-                pipelineFactory,
-                eventFactory,
-                externalPipelineId,
-                externalJobName,
-                parentBuildId: build.id,
-                parentBuilds,
-                causeMessage: `Triggered by sd@${pipelineId}:${currentJobName}`,
-                parentEventId: event.id
-            };
-
+            // Handle no-join case
+            // Note: current job can be "external" in nextJob's perspective
             /* CREATE AND START NEXT BUILD IF ALL 3 SCENARIOS ARE TRUE
              * 1. No join
              * 2. ([~D,B,C]->A) currentJob=D, nextJob=A, joinList(A)=[B,C]
              *    joinList doesn't include D, so start A
              * 3. ([~D,B,C]-> sd@123:A) currentJob=D, nextJob=sd@123:A, joinList(A)=[sd@111:B,sd@111:C]
              *    joinList doesn't include sd@111:D, so start A
              */
-            // Note: current job can be "external" in nextJob's perspective
-            const currentJobNotInJoinList = !joinListNames.includes(currentJobName) &&
-                !joinListNames.includes(`sd@${pipelineId}:${currentJobName}`);
-
-            // Handle no join case
             if (joinListNames.length === 0 || currentJobNotInJoinList) {
+                // Next build is internal
                 if (!isExternal) {
+                    const internalBuildConfig = {
+                        jobFactory,
+                        buildFactory,
+                        eventFactory,
+                        pipelineId,
+                        jobName: nextJobName,
+                        username,
+                        scmContext,
+                        build, // this is the parentBuild for the next build
+                        baseBranch: event.baseBranch || null,
+                        parentBuilds
+                    };
+
                     return createInternalBuild(internalBuildConfig);
                 }
 
-                return createExternalBuild(externalBuildConfig).then(async (externalEvent) => {
-                    const externalJobId = workflowGraph.nodes
-                        .find(n => n.name === nextJobName).id;
-                    const externalBuildId = externalEvent.builds
-                        .find(b => b.jobId === externalJobId).id;
-                    const externalBuild = await buildFactory.get({ id: externalBuildId });
-                    const masterParentBuilds = {};
-
-                    masterParentBuilds[externalEvent.pipelineId] = {
-                        eventId: externalEvent.id,
-                        jobs: { [externalJobName]: externalBuildId }
-                    };
-                    const nextJobToTrigger = workflowParser.getNextJobs(workflowGraph,
-                        { trigger: nextJobName, chainPR: pipeline.chainPR });
+                console.log('no join, not internal');
```

Comment:
you could use `screwdriver-logger` and do `logger.debug`  eg: https://github.com/screwdriver-cd/datastore-sequelize/pull/61/files

Reply:
Will remove console.log after tests are done